homeassistant/components/onvif/camera.py

```diff
@@ -411,8 +411,11 @@ def __init__(self, hass, config):
             req = media_service.create_type("GetSnapshotUri")
             req.ProfileToken = profiles[self._profile_index].token
 
-            snapshot_uri = await media_service.GetSnapshotUri(req)
-            self._snapshot = snapshot_uri.Uri
+            try:
+                snapshot_uri = await media_service.GetSnapshotUri(req)
+                self._snapshot = snapshot_uri.Uri
+            except ServerDisconnectedError as err:
+                _LOGGER.debug("Camera does not support GetSnapshotUri: %s", err)
```

Comment:
Is `ONVIFError` different than `ServerDisconnectedError` ? What can still raise `ONVIFError`  ?

Reply:
I think we need both exception. Here `ServerDisconnectedError`is caught because the sever reset the connection. But if the onvif web server can answer but can't reply the right thing, it will raise `ONVIFError` as I understand.